arch/arc/core/mpu/arc_mpu_v2_internal.h

```diff
@@ -204,33 +204,14 @@ void arc_core_mpu_configure_thread(struct k_thread *thread)
 
 #if defined(CONFIG_MPU_STACK_GUARD)
 #if defined(CONFIG_USERSPACE)
-	if ((thread->base.user_options & K_USER) != 0) {
-		/* the areas before and after the user stack of thread is
-		 * kernel only. These area can be used as stack guard.
-		 * -----------------------
-		 * |  kernel only area   |
-		 * |---------------------|
-		 * |  user stack         |
-		 * |---------------------|
-		 * |privilege stack guard|
-		 * |---------------------|
-		 * |  privilege stack    |
-		 * -----------------------
-		 */
-		if (_mpu_configure(THREAD_STACK_GUARD_REGION,
-			thread->arch.priv_stack_start - STACK_GUARD_SIZE,
-			STACK_GUARD_SIZE) < 0) {
-			LOG_ERR("thread %p's stack guard failed", thread);
-			return;
-		}
-	} else {
-		if (_mpu_configure(THREAD_STACK_GUARD_REGION,
-			thread->stack_info.start - STACK_GUARD_SIZE,
-			STACK_GUARD_SIZE) < 0) {
-			LOG_ERR("thread %p's stack guard failed", thread);
-			return;
-		}
-	}
+	/* no stack guard when USERSPACE is enabled, because
+	 * - no need to set mpu stack guard for user stack as it is surrounded
+	     by kernel memory.
+	 * - privileged stack is generated automatically without space
+	 *   for stack guard
+	 * - better to use hardware stack checking as mpu stack guard will
+	     consume MPU entry.
+	 */
```

Comment:
This #if now only has some comments inside -> suggest to change the ifdef and remove the else.
The comment could be moved/merged with the Kconfig option or so, as explanation why it's not supported.

Reply:
OK